assets/src/routes/routes.js

```diff
@@ -1,27 +1,33 @@
 import { Grade } from '@material-ui/icons'
 
-const routes = courseId => [
-  {
-    path: `/${courseId}/grades`,
-    title: 'Grade Distribution',
-    icon: Grade,
-    description: "See where your grade sits within the course grade distribution.",
-    image: "/static/images/grade_distribution_icon.png",
-  },
-  {
-    path: `/${courseId}/assignment`,
-    title: 'Assignment Planning',
-    icon: Grade,
-    description: "See what assignments have the greatest impact on your grade.",
-    image: "/static/images/assignments_planning_icon.png",
-  },
-  {
-    path: `/${courseId}/files`,
-    title: 'Files Accessed',
-    icon: Grade,
-    description: "See what files you and your peers are reading.",
-    image: "/static/images/file_access_trends_icon.png",
-  }
-]
+const routes = (courseId, activeViews) => {
+  const allViews = [
+    {
+      path: `/${courseId}/grades`,
+      title: 'Grade Distribution',
+      icon: Grade,
+      description: 'See where your grade sits within the course grade distribution.',
+      image: '/static/images/grade_distribution_icon.png',
+      viewCode: 'gd'
+    },
+    {
+      path: `/${courseId}/assignment`,
+      title: 'Assignment Planning',
+      icon: Grade,
+      description: 'See what assignments have the greatest impact on your grade.',
+      image: '/static/images/assignments_planning_icon.png',
+      viewCode: 'ap'
+    },
+    {
+      path: `/${courseId}/files`,
+      title: 'Files Accessed',
+      icon: Grade,
+      description: 'See what files you and your peers are reading.',
+      image: '/static/images/file_access_trends_icon.png',
+      viewCode: 'fa'
+    }
```

Comment:
Here is another thing I noticed if I disable say grades view in a course and I see that it is taken out from `index page` and `sidebar`. But if I go directly to grades page of the course like `/course/12233/grades` I see that display of the grades view. instead, it should display a message that says 'Grade Distribution view is hidden for this course.`

Reply:
Got it, that makes sense.